.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,18 @@
+name: golangci-lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+  pull_request:
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v1.2.1
+        with:
+          version: v1.29
```

Comment:
instead of having lint in github workflow, we should move this to semaphore as a block of procedure.

Reply:
part of JIRA: https://confluentinc.atlassian.net/jira/software/projects/METRICS/boards/173?selectedIssue=METRICS-2118 I will delete this later